rhttpc-client/src/main/scala/rhttpc/client/proxy/FailureResponseHandleStrategyChooser.scala

```diff
@@ -38,8 +38,7 @@ object SkipAll extends FailureResponseHandleStrategyChooser {
 
 case class BackoffRetry(initialDelay: FiniteDuration, multiplier: BigDecimal, maxRetries: Int) extends FailureResponseHandleStrategyChooser {
   override def choose(attemptsSoFar: Int, lastPlannedDelay: Option[FiniteDuration]): ResponseHandleStrategy = {
-    val retriesSoFar = attemptsSoFar - 1
-    if (retriesSoFar + 1 > maxRetries) {
+    if (attemptsSoFar > maxRetries) {
```

Comment:
Now you are checking if number of attempts is greater than max number of retries. It looks like mistake, because first attempt is not a retry... It will work only because attempts are zero indexed. I think that we have problem somewhere else. IMO we should pass here naturally indexed (starting from 1) `attemptsSoFar`. WDYT? If you will correct problem in a source I will gratefully merge it...

Reply:
I made it so that we pass to this function an argument indicating which retry are we starting, so we have them naturally indexed now as you wished, I think this is a straightforward solution and doesn't introduce unnecessary head-scratching with these needless variable or thinking about little things like: attempts so far meaning "retries so far",  "is first try a retry" etc.

Reply:
Nice! Indeed it will be easier to understand this logic. One last change is needed - IMO we should change `ReliableProxy.handleFailure` to during invocation of `choose` will use named param in this place: `choose(currentRetryAttempt = request.attempt, request.lastPlannedDelay)`. Will be also nice to leave there a comment that invocation attempt are zero-indexed so first attempt is just naturally ordered first retry attempt. WDYT?

Reply:
Ok done.